### DISCUSSION TITLE
Allow Prefect Flows to store a default context

### DISCUSSION THREAD

**Discussion by xuevin** [PrefectHQ/prefect#4303] [Ideas]
I was wondering if it is possible to set a default context when registering a flow. (similar to how Parameter can have a default)  I know that the client can submit a context but it would be a nice feature to also specify a persistent default context for every run.

**Comment by cicdw** [comment]
We generally encourage users to use `Parameter`s for data / information that they depend on; is there a reason you prefer context for your use case?

**Reply by xuevin** [reply]
Hi Chris. Right now we are using the context to store metadata about how a flow was constructed. This metadata does not affect the output of the flow but is accessed during a collection task. It would be nice to persist this metadata with each run but it is currently a manual copy paste.  Is there another way to store such general flow metadata?
